src/rust/engine/src/externs.rs

```diff
@@ -87,6 +87,10 @@ impl Externs {
     (self.val_for)(self.context, key)
   }
 
+  pub fn val_for_id(&self, id: Id) -> Value {
+    self.val_for(&Key::new_with_anon_type_id(id))
```

Comment:
Would be awesome to remove the `type_id` from `Value` if possible (and leaving it on `Key`)... looks like of the 2 or 3 places it is used, all of the cases would likely be OK with computing to a Key first.